internal/service/pushbasedlogexport/model.go

```diff
@@ -0,0 +1,67 @@
+package pushbasedlogexport
+
+import (
+	"context"
+
+	"go.mongodb.org/atlas-sdk/v20231115008/admin"
+
+	"github.com/hashicorp/terraform-plugin-framework-timeouts/resource/timeouts"
+	"github.com/hashicorp/terraform-plugin-framework/attr"
+	"github.com/hashicorp/terraform-plugin-framework/diag"
+	"github.com/hashicorp/terraform-plugin-framework/types"
+
+	"github.com/mongodb/terraform-provider-mongodbatlas/internal/common/conversion"
+)
+
+func NewTFPushBasedLogExport(ctx context.Context, projectID string, apiResp *admin.PushBasedLogExportProject, timeout *timeouts.Value) (*TFPushBasedLogExportRSModel, diag.Diagnostics) {
+	tfModel := &TFPushBasedLogExportRSModel{
+		ProjectID:  types.StringPointerValue(&projectID),
+		BucketName: types.StringPointerValue(apiResp.BucketName),
+		IamRoleID:  types.StringPointerValue(apiResp.IamRoleId),
+		PrefixPath: types.StringPointerValue(apiResp.PrefixPath),
+		CreateDate: types.StringPointerValue(conversion.TimePtrToStringPtr(apiResp.CreateDate)),
+		State:      types.StringPointerValue(apiResp.State),
+	}
+
+	links, diagnostics := types.ListValueFrom(ctx, linkObjectType, NewTFLinksModel(apiResp.GetLinks()))
+	if diagnostics.HasError() {
+		return nil, diagnostics
+	}
+
+	tfModel.Links = links
+
+	if timeout != nil {
+		tfModel.Timeouts = *timeout
+	}
+	return tfModel, nil
+}
+
+func NewPushBasedLogExportReq(plan *TFPushBasedLogExportRSModel) *admin.PushBasedLogExportProject {
```

Comment:
as it's not needed now outside of the package (e.g. no unit tests yet) we might want to keep it private until we really need it public.  same comment for the rest

Reply:
I'll leave as is for now if it's okay to reduce merge issues with follow up PRs, as it is it can sometimes become tricky to split these PRs.
If it makes it better, I won't merge this PR until I'm very close to merging the unit tests too